database/ram/rrddim_mem.c

```diff
@@ -2,18 +2,67 @@
 
 #include "rrddim_mem.h"
 
+static Pvoid_t rrddim_JudyHS_array = NULL;
+static netdata_rwlock_t rrddim_JudyHS_rwlock = NETDATA_RWLOCK_INITIALIZER;
+
+
+// ----------------------------------------------------------------------------
+// metrics groups
+
+STORAGE_METRICS_GROUP *rrddim_metrics_group_get(STORAGE_INSTANCE *db_instance __maybe_unused, uuid_t *uuid __maybe_unused) {
+    return NULL;
+}
+
+void rrddim_metrics_group_release(STORAGE_INSTANCE *db_instance __maybe_unused, STORAGE_METRICS_GROUP *smg __maybe_unused) {
+    // if(!smg) return; // smg may be NULL
+    ;
+}
+
 // ----------------------------------------------------------------------------
 // RRDDIM legacy data collection functions
 
-STORAGE_METRIC_HANDLE *rrddim_metric_init(RRDDIM *rd, STORAGE_INSTANCE *db_instance __maybe_unused) {
+STORAGE_METRIC_HANDLE *
+rrddim_metric_get_or_create(RRDDIM *rd, STORAGE_INSTANCE *db_instance __maybe_unused, STORAGE_METRICS_GROUP *smg __maybe_unused) {
+    STORAGE_METRIC_HANDLE *t = rrddim_metric_get(db_instance, &rd->metric_uuid, smg);
+    if(!t) {
+        netdata_rwlock_wrlock(&rrddim_JudyHS_rwlock);
+        Pvoid_t *PValue = JudyHSIns(&rrddim_JudyHS_array, &rd->metric_uuid, sizeof(uuid_t), PJE0);
+        fatal_assert(NULL == *PValue);
+        *PValue = rd;
```

Comment:
Like previous PRs, I ran coverity on this PR and we got few warnings, like this:

```sh
database/ram/rrddim_mem.c: 34 in rrddim_metric_get_or_create()
28             netdata_rwlock_wrlock(&rrddim_JudyHS_rwlock);
29             Pvoid_t *PValue = JudyHSIns(&rrddim_JudyHS_array, &rd->metric_uuid, sizeof(uuid_t), PJE0);
30             fatal_assert(NULL == *PValue);
31             *PValue = rd;
32             t = (STORAGE_METRIC_HANDLE *)rd;
33             netdata_rwlock_unlock(&rrddim_JudyHS_rwlock);
>>>     CID 358386:  Resource leaks  (RESOURCE_LEAK)
>>>     Variable "PValue" going out of scope leaks the storage it points to.
34         }
35     
36         if((RRDDIM *)t != rd)
37             fatal("RRDDIM_MEM: incorrect pointer returned from index.");
38     
39         return (STORAGE_METRIC_HANDLE *)rd;
```

Reply:
this is how judy works